packages/wrangler/src/dev.tsx

```diff
@@ -391,13 +397,36 @@ function useCustomBuild(
       stderr: "inherit",
       stdout: "inherit",
     });
+    if (watch_dir) {
+      watch(watch_dir, { persistent: true, ignoreInitial: true }).on(
+        "all",
+        (_event, _path) => {
+          console.log("Restarting build...");
+          cmd.kill();
+          cmd = execa(commandPieces[0], commandPieces.slice(1), {
+            ...(cwd && { cwd }),
+            stderr: "inherit",
+            stdout: "inherit",
+          });
```

Comment:
This block is now repeated - let's move it into its own function?

Reply:
It's just one statement, doesn't pass my threshold for a new abstraction yet imo